Enhancement(urls): UnifiedRouter::server WASM stub missing server_fn (and other native-only) builder methods, forcing user-side #[cfg(native)]

## Summary

Today, callers of `UnifiedRouter::new().server(|s| ...)` in cross-target code must wrap the `.server(...)` call (or its closure body) in `#[cfg(native)]` whenever the closure invokes a builder method that is **not** mirrored on `ServerRouterStub`. The most common such method is `ServerRouter::server_fn`, which `pages`-feature consumers reach for to register `#[server_fn]`-style endpoints.

The drift detection scaffold added in #4185 (`const _: fn() = || { ... };` in `crates/reinhardt-urls/src/routers/unified_router.rs`) already exercises ~22 builder methods on the stub, but `server_fn` is not included, so the stub does not have it and consumer code breaks on WASM.

The intent of `ServerRouterStub` (per its doc comment and #4185 drift detection rationale) is precisely to absorb the closure body so users can write a single `.server(|s| ...)` call that compiles cross-target. `server_fn` is the only common member missing from that surface today.

## Reproduction

```rust
use reinhardt::urls::routers::UnifiedRouter;

// Imagine `marker` is a `#[server_fn]`-emitted ZST marker on native.
fn build_router() -> UnifiedRouter {
    UnifiedRouter::new()
        .server(|s| s.server_fn(login_marker))
        .client(|c| c.named_route(\"home\", \"/\", home_page))
}
```

- `cargo check` (default native target): **OK**.
- `cargo check --target wasm32-unknown-unknown`: **FAIL** — `error[E0599]: no method named server_fn found for struct ServerRouterStub`.

The user's only recourse today is:

```rust
fn build_router() -> UnifiedRouter {
    let r = UnifiedRouter::new();
    #[cfg(native)]
    let r = r.server(|s| s.server_fn(login_marker));
    r.client(|c| c.named_route(\"home\", \"/\", home_page))
}
```

…which defeats the purpose of having `ServerRouterStub` at all.

## Proposal

Add the missing native-only builder methods (starting with `server_fn`) to `ServerRouterStub` as no-op stubs, and extend the `#4185` drift-detection `const _` to cover them.

Suggested patch sketch (`crates/reinhardt-urls/src/routers/unified_router.rs`):

```rust
#[cfg(wasm)]
impl ServerRouterStub {
    // ... existing stubs ...

    /// No-op stub for `ServerRouter::server_fn`.
    pub fn server_fn<M>(self, _marker: M) -> Self {
        self
    }
}

#[cfg(all(wasm, feature = \"client-router\"))]
#[doc(hidden)]
const _: fn() = || {
    let _ = UnifiedRouter::new()
        .server(|s| {
            s.with_prefix(\"/api\")
                // ... existing chain ...
                .server_fn(())   // <-- add to drift-detection
        })
        .client(|c| c);
};
```

Add similar stubs for any other native-only builder methods missing from the stub (audit `ServerRouter`'s public builder API for completeness).

## Why this matters

`reinhardt-cloud` (and any other consumer that uses `#[server_fn]` markers) hits this every time it tries to consolidate a router builder into a single cross-target function. Today the workaround is to wrap the entire builder in `#[cfg(native)]`, which means we cannot simply share router-construction code between WASM and native compilation paths — even though `UnifiedRouter` is explicitly designed to be a single cross-target builder.

## Tracking

A Reinhardt Cloud `upstream-tracking`-labelled issue will be filed once this is accepted (per Reinhardt Cloud's UR-4 convention).

## Related

- #4185 — drift detection scaffold for `ServerRouterStub`
- #4242 / #4234 — surrounding `pages::Router` deprecation work that motivated using `UnifiedRouter` as the single cross-target builder

🤖 Reported via [Claude Code](https://claude.com/claude-code) while migrating cloud-side workarounds.

--- a/crates/reinhardt-urls/src/routers/unified_router.rs
+++ b/crates/reinhardt-urls/src/routers/unified_router.rs
@@ -787,6 +787,20 @@ impl ServerRouterStub {
 	pub fn endpoint<F>(self, _f: F) -> Self {
 		self
 	}
+
+	/// No-op stub for `ServerFnRouterExt::server_fn` on the WASM side.
+	///
+	/// `server_fn` is provided to native `ServerRouter` by the
+	/// `ServerFnRouterExt` extension trait in `reinhardt-pages`. That trait
+	/// (and its `ServerFnRegistration` bound) is a native-only concern, so
+	/// the stub absorbs the call as a free-standing inherent method without
+	/// any trait bound. This lets cross-target builder chains such as
+	/// `UnifiedRouter::new().server(|s| s.server_fn(marker))` compile on
+	/// `wasm32-unknown-unknown` without `#[cfg(native)]` gates at the call
+	/// site.
+	pub fn server_fn<S>(self, _marker: S) -> Self {
+		self
+	}
 }
 
 #[cfg(wasm)]
@@ -831,6 +845,7 @@ const _: fn() = || {
 				.view_named("/v", "v", ())
 				.viewset("/vs/", ())
 				.endpoint(|| ())
+				.server_fn(())
 		})
 		.client(|c| c);
 };

